File: AetherPlayerShop/src/main/java/loyfael/listener/ShopConfiguration.java
package loyfael.listener;

import loyfael.model.Shop;
import loyfael.model.ShopItem;
import org.bukkit.Location;
import org.jetbrains.annotations.NotNull;

import java.util.UUID;

/**
 * Configuration temporaire d'un shop en cours de création
 */
public class ShopConfiguration {

    private final UUID playerUUID;
    private Location signLocation;
    private final Location chestLocation;
    private final ShopItem shopItem;
    private final int totalQuantity;

    private Shop.ShopType shopType = Shop.ShopType.SELL; // Par défaut = vente
    private double price = 1.0; // Prix par défaut

    public ShopConfiguration(@NotNull UUID playerUUID, @NotNull Location signLocation,
                           @NotNull Location chestLocation, @NotNull ShopItem shopItem,
                           int totalQuantity) {
        this.playerUUID = playerUUID;
        this.signLocation = signLocation;
        this.chestLocation = chestLocation;
        this.shopItem = shopItem;
        this.totalQuantity = totalQuantity;
    }

    // Getters
    public UUID getPlayerUUID() { return playerUUID; }
    public Location getSignLocation() { return signLocation; }
    public Location getChestLocation() { return chestLocation; }
    public ShopItem getShopItem() { return shopItem; }
    public int getTotalQuantity() { return totalQuantity; }
    public Shop.ShopType getShopType() { return shopType; }
    public double getPrice() { return price; }

    // Setters
    public void setShopType(@NotNull Shop.ShopType shopType) { this.shopType = shopType; }
    public void setPrice(double price) { this.price = Math.max(0.01, price); } // Min 1 centime
    public void setSignLocation(@NotNull Location signLocation) { this.signLocation = signLocation; }

    public void adjustPrice(double change) {
        setPrice(this.price + change);
    }
}

File: AetherPlayerShop2/src/main/kotlin/loyfael/listener/ShopConfiguration.kt
package loyfael.listener

import loyfael.model.Shop
import loyfael.model.ShopItem
import org.bukkit.Location
import java.util.*

/**
 * Configuration temporaire d'un shop en cours de création
 */
data class ShopConfiguration(
    val playerUUID: UUID,
    var signLocation: Location?,
    val chestLocation: Location,
    val shopItem: ShopItem,
    val totalQuantity: Int,
    var shopType: Shop.ShopType = Shop.ShopType.SELL,
    var teleportPolicy: Shop.TeleportPolicy = Shop.TeleportPolicy.ALLOW_TP
) {
    var price: Double = 1.0
        set(value) {
            field = maxOf(0.01, value) // Min 1 centime
        }

    fun adjustPrice(change: Double) {
        price = price + change
    }
}
